src/file/multi.ts

```diff
@@ -38,13 +35,15 @@ export class LinkedMetaPage {
   async metadata(): Promise<ArrayBuffer> {
     const pageData = await this.getMetaPage();
 
-    const lengthView = new DataView(pageData, 8 * N + 16);
-
-    // read the first four because that represents length
-    const metadataLength = lengthView.getUint32(0, true);
-    const start = 8 * N + 20;
+    const metadataView = new DataView(
+      pageData,
+      12 /* next pointer */ +
+        1 /* count */ +
+        this.index * (12 /* root mp */ + 1 /* length */ + 256) /* data */,
+    );
 
-    return pageData.slice(start, start + metadataLength);
+    const metadataLength = metadataView.getUint8(12);
```

Comment:
I'm a bit confused here as to why we do a `getUint8()` when the golang code allocates the four bytes for the metadata length: https://github.com/kevmo314/appendable/pull/245/files#diff-a262fc44f58b7959f07f93936218a158bb54f129aea0bf0516346ec26f671c2fR69-R84

Maybe I'm misunderstanding?

Reply:
The go code might be a mistake, can you change it to one byte? 4 bytes is overkill since the max length is 256. I thought I changed that but maybe not...